03Homeworkanswers.ipynb

```diff
@@ -0,0 +1,117 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Assignment 1\n",
+    "\n",
+    "Let's go back to our library program. You can use your solution to last weeks homework as starter code and build on it. Here is what the program should be able to do:\n",
+    "\n",
+    "From the previous assignment:\n",
+    "\n",
+    "- check out books from the library.\n",
+    "- each book can be checked out only once\n",
+    "- books can be returned to the library\n",
+    "- the program keeps track of who checked out a book\n",
+    "\n",
+    "For this assignment add the following features:\n",
+    "\n",
+    "- if a book is not in the library, users should be able to request them to be bought\n",
+    "- users can also donate books to the library\n",
+    "- the library also stores the authors of books and users can query for those (e.g. enter a title and the program returns the authors)\n",
+    "\n",
+    "Use functions to structure your code. \n",
+    "\n",
+    "There are many ways to write this program. It is up to you how you structure interaction with it (e.g. first ask for what a user wants to do, then ask for specific inputs or try to capture all cases with one input process).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "What book do you want?Lily\n",
+      "You can request our library to buy Lily\n"
+     ]
+    }
+   ],
+   "source": [
+    "# your code\n",
+    "book={\"Godfather\": \"ABC\", \"Harry Potter\": \"DEF\", \"Star Wars\": \"GHI\", \"On Justice\": \"JKL\"}\n",
+    "otherbook={\"Lily\": \"ABC\", \"Dog\": \"DEF\", \"Cat\": \"CHI\"}\n",
+    "\n",
+    "def new_books(item):\n",
+    "    item=input(\"Do your library have \"+ item)\n",
+    "def authors_books(item):\n",
+    "    item=input(\"Do your library have other books by \" + otherbook.values())\n",
+    "while True:\n",
+    "    item=input(\"What book do you want?\")\n",
+    "    if item==\"No\":\n",
+    "        break\n",
+    "\n",
+    "    if item in book:\n",
+    "        print(\"We have \" + item)\n",
+    "    if not item in book:\n",
+    "        print(\"You can request our library to buy \"+ item)\n",
+    "    elif item in otherbook:\n",
+    "        print(\"We have his otherbooks.\")"
```

Comment:
This doesn't implement the things asked for in the assignment:
- if a book is not in the library, users should be able to request them to be bought
- users can also donate books to the library
- the library also stores the authors of books and users can query for those (e.g. enter a title and the program returns the authors)